lib/instrumentation/reporting.spec.ts

```diff
@@ -0,0 +1,177 @@
+import type { S3Client } from '@aws-sdk/client-s3';
+import { mockDeep } from 'jest-mock-extended';
+import { s3 } from '../../test/s3';
+import { fs, logger } from '../../test/util';
+import type { RenovateConfig } from '../config/types';
+import type { PackageFile } from '../modules/manager/types';
+import type { BranchCache } from '../util/cache/repository/types';
+import {
+  addBranchStats,
+  addExtractionStats,
+  exportStats,
+  getReport,
+} from './reporting';
+
+jest.mock('../util/fs');
+jest.mock('../util/s3');
```

Comment:
```suggestion
jest.mock('../util/fs', ()=> mockDeep());
jest.mock('../util/s3', ()=> mockDeep());
```

So s3 is never loaded, by default jest will load them recursively to create proper mocks. When we pass a mock factory, then jest will skip loading those files.